lib/matplotlib/tests/test_legend.py

```diff
@@ -1450,3 +1451,53 @@ def test_boxplot_labels():
     handles, labels = ax.get_legend_handles_labels()
     assert len(handles) == 0
     assert len(labels) == 0
+
+
+def test_boxplot_legend_labels():
+    # Test that legend entries are generated from boxplots. `label` is the new
+    # parameter for setting legend labels.
+    np.random.seed(19680801)
+    data = np.random.lognormal(size=(68, 4), mean=1.5, sigma=0.2)
+    labels = list('ABCD')
+    colors = ['peachpuff', 'orange', 'tomato', 'plum']
+
+    legend_labels = ['peaches', 'oranges', 'tomatoes', 'plums']
+    fig, axs = plt.subplots(nrows=2,
+                            ncols=3,
+                            figsize=(10, 7),
+                            sharey=True)
+    # This plot will not generate a legend.
+    first = axs[0, 0].boxplot(data, patch_artist=True)
+    with pytest.warns(UserWarning):
+        axs[0, 0].legend()
+    empty_list = axs[0, 0].get_legend_handles_labels()
+    assert not empty_list[0] and not empty_list[1]
+
+    # Testing legend labels and patch passed to legend.
+    second = axs[0, 1].boxplot(data, patch_artist=True, label=legend_labels)
+    for i, v in enumerate(second['boxes']):
+        assert v.get_label() == legend_labels[i]
+
+    # Testing legend without `box`.
+    third = axs[0, 2].boxplot(data, label=legend_labels, showbox=False)
+    # Without a box, The legend entries should be passed from the medians.
+    for i, v in enumerate(third['medians']):
+        assert v.get_label() == legend_labels[i]
+
+    # compute the boxplot stats
+    stats = cbook.boxplot_stats(data, labels=labels, bootstrap=10000)
+
+    # The next 2 plots are called with `Axes.bxp`
```

Comment:
Why?

Some remarks on test strategies for nested functions: Testing parts of the functionality on `bxp()` and other parts on `boxplot()` is rather confusing for the reader: Is that special behavior of only that function or a coincidental choice of `bxp()` vs `boxplot()`.

You could test the full behavior on `bxp()` and `boxplot()` but that's quite tedious. Instead, I propose to only test the outer `boxplot()` and rely that it's forwarding to `bxp()` so that it's implicitly tested as well. If you want to be very sure and completely rely on forwadings, add a single duplicated test for `bxp()` to make sure labels are basically handled inside bxp.